client/components/content-containers/tcc-exam/edit/Assessment.vue

```diff
@@ -0,0 +1,63 @@
+<template>
+  <assessment-item
+    @selected="expanded = !expanded"
+    @save="save"
+    @delete="$emit('delete')"
+    :assessment="assessment"
+    :expanded="expanded"
+    :draggable="true">
+    <template slot="header" slot-scope="{ isEditing }">
+      <v-container v-if="objectives.length" class="pa-0 mt-2">
+        <v-layout justify-end>
+          <v-flex xs4>
+            <multiselect
+              v-model="objective"
+              :options="objectives"
+              :searchable="true"
+              :disabled="!isEditing"
+              :track-by="'id'"
+              :custom-label="it => it.data ? it.data.name : ''"
+              :placeholder="objectiveLabel" />
+          </v-flex>
+        </v-layout>
+      </v-container>
+    </template>
+  </assessment-item>
+</template>
+
+<script>
+import AssessmentItem from '@/components/editor/structure/AssessmentItem';
+import find from 'lodash/find';
+import get from 'lodash/get';
+import Multiselect from '@/components/common/Select';
+import set from 'lodash/set';
+
+export default {
+  props: {
+    assessment: { type: Object, required: true },
+    objectives: { type: Array, required: true },
+    objectiveLabel: { type: String, required: true }
+  },
+  data() {
+    return {
+      expanded: !this.assessment.id,
+      objective: null
+    };
+  },
+  methods: {
+    save(assessment) {
+      set(assessment, 'refs.objectiveId', get(this.objective, 'id', null));
+      this.$emit('save', assessment);
+    }
+  },
+  mounted() {
+    const objectiveId = get(this.assessment, 'refs.objectiveId');
+    if (!objectiveId) return;
```

Comment:
`objective` is `null` by default. This is only called once since it is in the `mounted` hook so there should be no need for handling a situation where the `objective` has value and should be reset to `null`.

Reply:
You're right, I had (for some weird reason) `updated` hook in mind.
Looking again I would say that this shouldn't be in `mounted` hook but in `created`. Changing data (synchronously) in `mounted` hook forces Vue to unnecessarily re-render the component one more time right away.